middleware/expandDocument.js

```diff
@@ -0,0 +1,72 @@
+const dotProp = require('dot-prop');
+const _ = require('lodash');
+
+function service(esclient) {
+  function expandDocument(doc) {
+    const traverse = (obj, cb, keypath) => {
+      for (let k in obj) {
+        const newkeypath = keypath ? keypath + '.' + k : k;
+
+        // don't recurse down arrays so that we can send them to ES all at once
+        if (obj[k] && typeof obj[k] === 'object' && !Array.isArray(obj[k])) {
+          traverse(obj[k], cb, newkeypath);
+        } else {
+          cb(obj[k], newkeypath);
+        }
+      }
+    };
+
+    const promises = [];
+    doc.debug = doc.debug || {};
+    doc.debug.expanded = {};
+
+    const lookupCb = (value, keypath) => {
+      if (!value || typeof value === 'number') {
```

Comment:
Should this validation should be checking "string or array of strings"?

> text
> (Required, string or array of strings) Text to analyze. If an array of strings is provided, it is > analyzed as a multi-value field.

https://www.elastic.co/guide/en/elasticsearch/reference/7.4/indices-analyze.html

Reply:
changed + commented